build_tools/benchmarks/suites/cmake_rule_generator.py

```diff
@@ -150,19 +162,89 @@ def add_import_model_rule(
       )
 
     import_model_rule = IreeModelImportRule(target_name=target_name,
+                                            model_id=model_id,
+                                            model_name=model_name,
                                             output_file_path=output_file_path,
                                             mlir_dialect_type=mlir_dialect_type,
                                             cmake_rule=cmake_rule)
 
     self._import_model_rules[model_id] = import_model_rule
     return import_model_rule
 
+  def add_compile_module_rule(self,
+                              compile_config: iree_definitions.CompileConfig,
+                              model_import_rule: IreeModelImportRule):
+    """Adds a rule to compile a MLIR into a IREE module. Reuses the existing
+    rule when possible."""
+
+    model_id = model_import_rule.model_id
+    model_name = model_import_rule.model_name
+
+    target_id = f"{model_id}-{compile_config.id}"
+    if target_id in self._compile_module_rules:
+      return self._compile_module_rules[target_id]
+
+    # Module target: <package_name>_iree-module-<model_id>-<compile_config_id>
+    target_name = f"iree-module-{target_id}"
+
+    # Module path: <iree_artifacts_dir>/<model_id>_<model_name>/<compile_config_id>.vmfb
+    output_path = f"{self._iree_artifacts_dir}/{model_id}_{model_name}/{compile_config.id}.vmfb"
```

Comment:
Use `os.path.join` instead of custom path manipulation

```suggestion
    output_path = os.path.join(self._iree_artifacts_dir,
                               f"{model_id}_{model_name},
                               f"{compile_config.id}.vmfb")
```